src/service/vulnerabilities/rag_poisoning_lab.py

```
@@ -0,0 +1,489 @@
+        \"https://external-api.com/process\",
+        headers=headers,
+        data=request_body
+    )
+    
+    logger.info(f\"External response: {external_response.json()}\")
+    return external_response.json()
+'''
+    return ""
+
+
+def generate_malicious_logs(level: int, user_agent: str = "Mozilla/5.0", auth_token: str = "Bearer sk-123abc456") -> List[Dict]:
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to function default: "auth_token"

(S107)